storey/flow.py

```diff
@@ -1552,7 +1594,124 @@ def _static_run(*args, **kwargs):
     return _sval._run(*args, **kwargs)
 
 
-class ParallelExecution(Flow):
+class Executor:
+    def __init__(
+        self,
+        max_processes: Optional[int] = None,
+        max_threads: Optional[int] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.runnables: dict[str, ParallelExecutionRunnable] = {}
+        self._runnable_by_name = {}
+
+        self.max_processes = max_processes or os.cpu_count() or 16
+        self.max_threads = max_threads or 32
+        self.num_processes = 0
+        self.num_threads = 0
+
+        self._process_executor_by_runnable_name = {}
+        self._mp_context = None
+
+    def add_runnable(self, runnable: ParallelExecutionRunnable) -> None:
+        if runnable.name not in self.runnables:
+            self.runnables[runnable.name] = runnable
+        else:
+            raise ValueError(f"Model name '{runnable.name}' is not unique")
```

Comment:
```suggestion
            raise ValueError(f"ParallelExecutionRunnable name '{runnable.name}' is not unique")
```

Reply:
Also, don't we have the exact same check in `init_runnable`? Do we really need to check it in both the constructor and `_init()`?